statistic/statistics.py

```diff
@@ -106,66 +86,61 @@ def render(tpl_path, context):
+    Returns the list of paths of YANG Modules (.yang) in all sub-directories
         Arguments
             :param srcdir: (str) root directory to search for yang files
             :return: list of YANG files
     """
     ll = []
     for root, _, files in os.walk(srcdir):
         for f in files:
-            if f.endswith(".yang"):
+            if f.endswith('.yang'):
                 ll.append(os.path.join(root, f))
     return ll
 
 
-def get_specifics(path_dir):
-    """Get amount of yang files in specified directory and the amount that
-    passed compilation
+def get_total_and_passed(dir: str) -> t.Tuple[int, int]:
+    """Get the number of yang files in a specified directory and the
+    number that passed compilation.
         Arguments:
-            :param path_dir: (str) path to directory where we are searching for
+            :param path_dir: (str) path to the directory where to search for
                 yang files
-            :return: list containing amount of yang files and amount that pass
-                compilation respectively
+            :return: tuple containing the number of yang files and the number
+                that passed compilation respectively
     """
     passed = 0
     num_in_catalog = 0
-    yang_modules = list_of_yang_modules_in_subdir(path_dir)
-    yang_modules_length = len(yang_modules)
-    x = 0
+    yang_modules = list_yang_modules_recursive(dir)
+    num_of_modules = len(yang_modules)
     checked = {}
-    for mod_git in yang_modules:
-        x += 1
-        LOGGER.info("{} out of {} getting specifics from {}".format(x, yang_modules_length, path_dir))
-        m_name_revision = mod_git.split('/')[-1]
-        if m_name_revision in checked.keys():
-            if checked[m_name_revision]['passed']:
-                passed += 1
-            if checked[m_name_revision]['in-catalog']:
-                num_in_catalog += 1
+    for i, module_path in enumerate(yang_modules, start=1):
+        LOGGER.info('{} out of {} getting specifics from {}'.format(i, num_of_modules, dir))
+        filename = os.path.basename(module_path)
+        if filename in checked.keys():
+            passed += checked[filename]['passed']
+            num_in_catalog += checked[filename]['in-catalog']
             continue
-        checked[m_name_revision] = {}
-        checked[m_name_revision]['passed'] = False
-        checked[m_name_revision]['in-catalog'] = False
-        try:
-            parsed_yang = yangParser.parse(os.path.abspath(mod_git))
-            revision = parsed_yang.search('revision')[0].arg
-        except:
-            continue
-        organization = resolve_organization(mod_git, parsed_yang)
-        name = m_name_revision.split('.')[0].split('@')[0]
+        checked[filename] = {}
+        checked[filename]['passed'] = False
+        checked[filename]['in-catalog'] = False
+        revision = None
+        if parsed_yang := yangParser.parse(os.path.abspath(module_path)):
```

Comment:
**TYPO**
 := instead of == 
 in various places in script

Reply:
_"parsed_yang"_ is not defined

Reply:
:= is not a typo, it's a new operator introduced in python 3.8. https://docs.python.org/3/reference/expressions.html#assignment-expressions In each case it only really saves one line of code. If we want to avoid it in our codebase, I can remove it.

Reply:
ou sorry, I wasn't familiar with that ... I would probably avoid that for now - we can add this in the future

Reply:
No problem, done.